src/test/scala/ch/epfl/bluebrain/nexus/tests/NewBaseSpec.scala

```diff
@@ -0,0 +1,194 @@
+package ch.epfl.bluebrain.nexus.tests
+
+import java.util.regex.Pattern.quote
+
+import akka.http.javadsl.model.headers.HttpCredentials
+import akka.http.scaladsl.model.HttpMethods._
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.headers.Authorization
+import akka.http.scaladsl.testkit.ScalatestRouteTest
+import akka.http.scaladsl.unmarshalling.FromEntityUnmarshaller
+import ch.epfl.bluebrain.nexus.commons.http.HttpClient
+import ch.epfl.bluebrain.nexus.commons.http.HttpClient.UntypedHttpClient
+import ch.epfl.bluebrain.nexus.commons.http.JsonLdCirceSupport._
+import ch.epfl.bluebrain.nexus.commons.test.Resources
+import ch.epfl.bluebrain.nexus.tests.config.TestConfig
+import ch.epfl.bluebrain.nexus.tests.config.TestConfig.RemoteDeltaConfig
+import ch.epfl.bluebrain.nexus.tests.iam.types.{AclEntry, AclListing, AclUser}
+import com.typesafe.config.ConfigFactory
+import io.circe.Json
+import monix.bio.Task
+import monix.execution.Scheduler.Implicits.global
+import org.scalatest.{Assertion, BeforeAndAfterAll}
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpecLike
+
+import scala.concurrent.duration._
+
+trait NewBaseSpec extends AnyWordSpecLike
+  with BeforeAndAfterAll
+  with Resources
+  with ScalatestRouteTest
+  with ScalaFutures
+  with Matchers {
+
+  private[tests] val config: TestConfig = TestConfig.load(ConfigFactory.load())
+
+  private[tests] val realmLabel = "internal"
+
+  private[tests] implicit val cl: UntypedHttpClient[Task] = HttpClient.untyped[Task]
+
+  private def toAuthorizationHeader(token: String) =
+    Authorization(
+      HttpCredentials.createOAuth2BearerToken(token)
+    )
+
+  // Initializing the requirements according to the configuration
+  val (baseUrl: String, userTokens: Map[Identity, Authorization]) =
+    config match {
+      case r: RemoteDeltaConfig =>
+        // Urls and tokens are provided by the configuration
+        (
+          r.deltaUri,
+          r.userTokens.map { case (k, v) =>
+            User(k, "no_password") -> toAuthorizationHeader(v)
+          } ++ Map(Identity.ServiceAccount -> toAuthorizationHeader(r.serviceAccountToken))
+        )
+      case _ =>
+        // Using docker-compose, urls are available as system-properties
+        // And we retrieve tokens from the Keycloak that has been started
+        (
+          s"http://${System.getProperty("delta:8080")}/v1",
```

Comment:
`System.getProperty("delta:8080")` could be at the TestConfig

Reply:
Same thing